### PR TITLE
feat(AWS Deploy): Support `disableRollback` parameter

### PR DESCRIPTION
**Pull request by fredericbarthelet**
Closes: #10235

### CODE REVIEW

File: lib/plugins/aws/lib/updateStack.js [hunk 1]
```diff
@@ -62,6 +62,11 @@ module.exports = {
       params.Parameters = this.serverless.service.provider.stackParameters;
     }
 
+    if (this.serverless.service.provider.disableRollback) {
```
Comment (pgrzesik) [inline lib/plugins/aws/lib/updateStack.js:65]: I believe similar set of changes should be applied to `createStack` logic, what do you think?

File: lib/plugins/aws/provider.js [hunk 2]
```diff
@@ -844,6 +844,7 @@ class AwsProvider {
             },
             deploymentPrefix: { type: 'string' },
             disableDefaultOutputExportNames: { const: true },
+            disableRollback: { const: true },
```
Comment (pgrzesik) [inline lib/plugins/aws/provider.js:847]: It would be great to add that to `serverless.yml.md` reference file as well

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thanks @fredericbarthelet - looks good, I only have minor remarks, let me know what do you think 🙇

File: lib/plugins/aws/lib/updateStack.js (hunk 1 above)
Comment (fredericbarthelet) [reply lib/plugins/aws/lib/updateStack.js:65]: Agreed, I added it

**Review by pgrzesik — APPROVED** [review]
Well done, thank you @fredericbarthelet :bow:

File: docs/providers/aws/guide/serverless.yml.md [hunk 3]
```diff
@@ -261,6 +261,7 @@ provider:
   stackParameters:
     - ParameterKey: 'Keyname'
       ParameterValue: 'Value'
+  disableRollback: true # To be used for non-production environment
```
Comment (RichardBradley) [inline docs/providers/aws/guide/serverless.yml.md:264]: Why did you write "for non-production environment" here?

I think this param is to control a choice as to whether to attempt to fix a partial deployment, or to roll it all back. That's not necessarily tied to production or not.

"true" is required to work around AWS issues like https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/573 , whether in production or non-production

**Pull request merged**